crypto/x509v3/v3_ncons.c

```diff
@@ -297,47 +297,152 @@ int NAME_CONSTRAINTS_check(X509 *x, NAME_CONSTRAINTS *nc)
 
 }
 
+#define CN2DNSID_NOMEM   (-2)
+#define CN2DNSID_BADNAME (-1)
+
+static int cn2dnsid(ASN1_STRING *cn, unsigned char **dnsid)
+{
+    int utf8_length;    /* Return type of ASN1_STRING_to_UTF8 */
+    int i;
+    unsigned char *utf8_value;
+    int isdnsname = 0;
+
+    /*
+     * Per RFC 6125 DNS-IDs representing internationalized domain names appear
+     * in certificates in A-label encoded form.  The same applies to CNs which
+     * are intended to represent DNS names.  However, while in the SAN DNS-IDs
+     * are IA5Strings as CNs they may be needlessly stored in 16-bit Unicode.
+     * We perform a conversion to UTF-8 here, to ensure that we get an ASCII
+     * representation of strings that have been encoded in some other way in
```

Comment:
"ASCII representation" is misleading in the context. I mean ASCII implies 7-bit chars, and ASCII representation of Unicode is UTF-7, not UTF-8. Well, one can always say "you know what I mean", and everybody kind of does, so this would be a side note... Otherwise one can probably use "'char' representation" (when 'char' refers to C type). or "C string representation"...